### DISCUSSION TITLE
lxqt-powermanagement crashes after Qt upgrade (in Arch)

### DISCUSSION THREAD

**Discussion by tobrunn** [lxqt/lxqt#2219] [General]
Since update  qt5-base (5.15.2+kde+r302-1 -> 5.15.3+kde+r137-1) lxqt-powermanagement regulary crashes repeatedly after a while with segfault and core-dump. The screen brightness gets lower with every crash till screen is nearly completely dark.
A window appears 'Crash report: lxqt-powermanagement crashed too many times, disabling'.

The error message "error 4 in libQt5Core.so.5.15.3" seems to be related / could be a regession of [issue#925](https://github.com/lxqt/pcmanfm-qt/issues/925)? 

##### Expected Behavior

lxqt-powermanagement should not crash
##### Current Behavior

see above
##### Possible Solution

none found yet but disabling the 'reduce backlight brightness' feature seems to prevent crashes
##### Steps to Reproduce (for bugs)

1.  Boot your device (here ThinkPad 11e 20LQ)
2.  Wait some time till backlight brightness reduces and crash report appears.
3. Re-enable lxqt-powermanagement in session-control / session-configuration -> basic-settings
4. wait for the next crash...

##### System Information

* Distribution & Version: ArchLinux
* Kernel: Linux TPe11 5.16.15-arch1-1  x86_64 GNU/Linux
* Qt Version: 5.15.3+kde+r137-1
* liblxqt Version: liblxqt 1.0.0-1

[coredump_lxqt_pm.txt](https://github.com/lxqt/lxqt/files/8314919/coredump_lxqt_pm.txt)

**Comment by tsujan** [comment]
Thanks for the report!

A crash after a Qt upgrade isn't necessarily a sign of a bug; it can simply show that some libraries need recompilation.

> seems to be related / a regession of issue#925

No! they aren't related.

I still have `qt5-base-5.15.3+kde+r136-1` and, of course, see no crash. If you can recompile `libqtxdg`, `liblxqt`, `lxqt-qtplugin` and `lxqt-powermanagement`, please do so and test by restarting "Power Management" from LXQt Session Setting (the order is important and not all of them may need recompilation — test one by one). If not, please be patient.

**Comment by tsujan** [comment]
Sadly, the same thing may be true about KF5: it may also need a recompilation or even an upgrade.

**Comment by stefonarch** [comment]
I see that it's some days here that this version is arrived with a full upgrade, but didn't notice any coredumps.
`[2022-03-19T08:32:35+0100] [ALPM] upgraded qt5-base (5.15.3+kde+r133-1 -> 5.15.3+kde+r137-1)
`

**Comment by stefonarch** [comment]
Hm I updated now and saw now the message "...crashed too many times, disabled until next login" for the first time, bit there was no qt5 related update. Will investigate.

**Comment by tsujan** [comment]
Here, in Manjaro Testing, the update came last night as `5.15.3+kde+r134-1 -> 5.15.3+kde+r136-1`, without problem. But I'd recompiled `libqtxdg` for another reason.

**Comment by stefonarch** [comment]
It crashed again here too for the second time, after maybe half hour of work. 
[coredump.lxqt-powermanagment.txt](https://github.com/lxqt/lxqt/files/8318211/coredump.lxqt-powermanagment.txt)

**Comment by tsujan** [comment]
Please attach what comes after `where`:

```
coredumpctl gdb APP_NAME
where
```

**Comment by tsujan** [comment]
The crash happens in `kidletime`. Since the latest version in Arch is `5.92.0-1` and because the same version works fine with `qt5base-5.15.3+kde+r136-1`, the problem should be in `kidletime`.

Either recompile it and retry or wait until it's upgraded.

**Comment by stefonarch** [comment]
I ran it from command line:  

```
$ lxqt-powermanagement 
BatteryChanged discharging: false chargeLevel: 100 actionTime: QTime(Invalid)
Inhibit got: 31
Starting idlenesswatcher
"PowerButton" "" ""
"SuspendButton" "" ""
"HibenateButton" "" ""
BatteryChanged discharging: true chargeLevel: 100 actionTime: QTime(Invalid)
QString::arg: Argument missing: "43 minuti" , 43
QString::arg: Argument missing: "2 ore" , 2

[....]

QString::arg: Argument missing: "3 ore" , 3
QString::arg: Argument missing: "46 minuti" , 46
QString::arg: Argument missing: "3 ore" , 3
BatteryChanged discharging: true chargeLevel: 79.5299 actionTime: QTime(Invalid)
Errore di segmentazione (core dump creato)
```

Downgraded kidletime atm: `downgraded kidletime (5.92.0-1 -> 5.91.0-1)
`

**Comment by tsujan** [comment]
> Downgraded kidletime atm

I don't think that'll make a difference. If it needs recompilation, downgrading won't work. If there's a change in Qt, the problem exists in previous versions of `kidletime` too.

**Comment by stefonarch** [comment]
I've no clue yet, It's 2 days I had no crash until today (uptime is  now 1day). I should have seen  at least one yesterday evening or this morning but didn't. The first crash came after today's update, but I cannot see anything here.
```
[2022-03-21T13:20:26+0100] [ALPM] upgraded tzdata (2021e-1 -> 2022a-1)
[2022-03-21T13:20:26+0100] [ALPM] upgraded bluez (5.64-1 -> 5.64-2)
[2022-03-21T13:20:26+0100] [ALPM] upgraded bluez-libs (5.64-1 -> 5.64-2)
[2022-03-21T13:20:26+0100] [ALPM] upgraded bluez-utils (5.64-1 -> 5.64-2)
[2022-03-21T13:20:27+0100] [ALPM] upgraded python-feedparser (6.0.2-1 -> 6.0.3-1)
[2022-03-21T13:20:27+0100] [ALPM] upgraded python-pyparsing (3.0.2-1 -> 3.0.3-1)
[2022-03-21T13:20:27+0100] [ALPM] upgraded python-setuptools (1:59.3.0-1 -> 1:59.4.0-1)
[2022-03-21T13:20:27+0100] [ALPM] upgraded gnutls (3.7.3-1 -> 3.7.4-1)
[2022-03-21T13:20:27+0100] [ALPM] upgraded vulkan-icd-loader (1.3.207-1 -> 1.3.208-1)
[2022-03-21T13:20:27+0100] [ALPM] upgraded calibre (5.39.0-1 -> 5.39.1-1)
[2022-03-21T13:20:27+0100] [ALPM] upgraded chromaprint (1.5.1-2 -> 1.5.1-3)
[2022-03-21T13:20:28+0100] [ALPM] upgraded chromium (99.0.4844.74-1 -> 99.0.4844.82-1)
[2022-03-21T13:20:29+0100] [ALPM] upgraded fluidsynth (2.2.5-1 -> 2.2.6-1)
[2022-03-21T13:20:30+0100] [ALPM] upgraded imagemagick (7.1.0.27-1 -> 7.1.0.28-1)
[2022-03-21T13:20:30+0100] [ALPM] upgraded libsynctex (2021.59745-1 -> 2021.59745-2)
[2022-03-21T13:20:30+0100] [ALPM] upgraded man-db (2.10.1-1 -> 2.10.2-1)
[2022-03-21T13:20:30+0100] [ALPM] upgraded python-tomli (2.0.0-1 -> 2.0.1-1)
[2022-03-21T13:20:30+0100] [ALPM] upgraded python-urllib3 (1.26.8-1 -> 1.26.9-1)
[2022-03-21T13:20:31+0100] [ALPM] upgraded qt6-translations (6.2.3-1 -> 6.2.4-1)
[2022-03-21T13:20:32+0100] [ALPM] upgraded qt6-base (6.2.3-5 -> 6.2.4-1)
[2022-03-21T13:20:32+0100] [ALPM] upgraded qt6-5compat (6.2.3-2 -> 6.2.4-1)
[2022-03-21T13:20:33+0100] [ALPM] upgraded qt6-declarative (6.2.3-2 -> 6.2.4-1)
[2022-03-21T13:20:33+0100] [ALPM] upgraded qt6-imageformats (6.2.3-2 -> 6.2.4-1)
[2022-03-21T13:20:33+0100] [ALPM] upgraded qt6-positioning (6.2.3-2 -> 6.2.4-1)
[2022-03-21T13:20:33+0100] [ALPM] upgraded qt6-quicktimeline (6.2.3-2 -> 6.2.4-1)
[2022-03-21T13:20:33+0100] [ALPM] upgraded qt6-quick3d (6.2.3-3 -> 6.2.4-1)
[2022-03-21T13:20:33+0100] [ALPM] upgraded qt6-serialport (6.2.3-2 -> 6.2.4-1)
[2022-03-21T13:20:33+0100] [ALPM] upgraded qt6-shadertools (6.2.3-2 -> 6.2.4-1)
[2022-03-21T13:20:33+0100] [ALPM] upgraded qt6-svg (6.2.3-2 -> 6.2.4-1)
[2022-03-21T13:20:34+0100] [ALPM] upgraded qt6-tools (6.2.3-4 -> 6.2.4-1)
[2022-03-21T13:20:34+0100] [ALPM] upgraded qt6-webchannel (6.2.3-2 -> 6.2.4-1)
[2022-03-21T13:20:40+0100] [ALPM] upgraded qt6-webengine (6.2.3-8 -> 6.2.4-1)
```

**Comment by palinek** [comment]
> coredump.lxqt-powermanagment.txt

```
                                               Stack trace of thread 107616:
                                               #0  0x00007fcb52539a09 QHashData::nextNode(QHashData::Node*) (libQt5C>
                                               #1  0x00007fcb53d274c9 n/a (libKF5IdleTime.so.5 + 0x44c9)
                                               #2  0x00007fcb526da463 n/a (libQt5Core.so.5 + 0x2bd463)
                                               #3  0x00007fcb53d27226 AbstractSystemPoller::timeoutReached>
                                               #4  0x00007fcb4c2b871d n/a (KF5IdleTimeXcbPlugin0.so + 0x471d)
                                               #5  0x00007fcb5269fef7 _ZN24QAbstractEventDispatcher17filterNativ>
                                               #6  0x00007fcb4e8b4b89 _ZN14QXcbConnection14handleXcbEventEP19xcb>
                                               #7  0x00007fcb4e8b8d5c _ZN14QXcbConnection16processXcbEventsE6QFl>
                                               #8  0x00007fcb4e8dbf58 n/a (libQt5XcbQpa.so.5 + 0x63f58)
                                               #9  0x00007fcb539a7ee3 g_main_context_dispatch (libglib-2.0.so.0 >
                                               #10 0x00007fcb539fe0f9 n/a (libglib-2.0.so.0 + 0xab0f9)
                                               #11 0x00007fcb539a5455 g_main_context_iteration (libglib-2.0.so.0>
                                               #12 0x00007fcb526f557a _ZN20QEventDispatcherGlib13processEventsE6>
                                               #13 0x00007fcb526a188b _ZN10QEventLoop4execE6QFlagsINS_17ProcessE>
                                               #14 0x00007fcb526acfd7 QCoreApplication::exec() (libQt5Core.s>
                                               #15 0x000055cbc265e945 main (lxqt-powermanagement + 0x1a945)
                                               #16 0x00007fcb51f15310 __libc_start_call_main (libc.so.6 + 0x2d31>
                                               #17 0x00007fcb51f153c1 __libc_start_main@@GLIBC_2.34 (libc.so.6 +>
                                               #18 0x000055cbc2650305 _start (lxqt-powermanagement + 0xc305)
```

It doesn't seem, that LXQt is involved in this problem...

**Comment by stefonarch** [comment]
After downgrading kidletime I've seen no crash anymore, but as they were only three before it's early to say.

**Comment by tsujan** [comment]
> It doesn't seem, that LXQt is involved in this problem...

Most probably, it'll be fixed by a future upgrade.

**Comment by stefonarch** [comment]
Just for the record: still no crash with downgraded ` kidletime 5.91.0-1` on arch.

**Comment by tsujan** [comment]
> Just for the record: still no crash with downgraded kidletime 5.91.0-1 on arch.

Have you also enabled baclkight change?

There may be a reason that qt5-base is still `5.15.3+kde+r136-1` in Manjaro Testing. I had an upgrade an hour ago but it wasn't upgraded to ` 5.15.3+kde+r137-1`.

**Comment by stefonarch** [comment]
> Have you also enabled backlight change?

Yes, after 4min idleness

**Comment by stefonarch** [comment]
Just now, using the "family" pc with git and updated arch, I see that power-management doesn't crash with those packages.
It's not visible in the panel because there is no battery, does idleness suspend, module is running 21 hours atm.

**Comment by tsujan** [comment]
I have qt5-base 5.15.3+kde+r137-1 for two days and there's no crash. Closing....

**Comment by stefonarch** [comment]
So then I removed yesterday `kidletime` from Ignorepkg and updated - just now instead of going in standby it crashed again.

**Comment by tsujan** [comment]
If it's the same crash — which most probably is — it isn't caused by `lxqt-powermanagement`. I don't see any crash with `kidletime-5.92.0-1` and `qt5-base-5.15.3+kde+r137-1`, both of which are from Arch but under Manjaro. So, I suspect a temporary inconsistency in Arch.

I also think this can be converted to a Discussion.

**Reply by stefonarch** [reply]
We should have a category, kind of "3rd party issues", ideas?

My PC (with arch too)  has still no issue.

**Reply by tsujan** [reply]
Since this is a question ("Why does it happen to some?") , I think the Q&A category is good.

**Reply by stefonarch** [reply]
Not sure about Q&A

There is some related issue: "suspend if idle for 20min" _sometimes_ doesn't work on the main PC (without battery). 
It's  an old thing, I never managed to really track down the conditions.

**Reply by tsujan** [reply]
There's nothing special about 20m, as far as I know. If it works with shorter periods, probably something happens before 20m, so that the computer can't be considered idle.

**Reply by stefonarch** [reply]
20m is just the actual config, it did fail also with15min. I tried once to do tests with shorter times like 1min and it always worked, it's random. Happens that I find it still on after hours.

**Comment by gorrioncillo** [comment]
The same occur on Lubuntu 22.04 after an upgrade from 21.10.
It crashes when the "backlight change" happen. It keep lowing down the brightness and  the error appear.
I try whit 0.17.0 version, and 1.1.0 from backports of lubuntu

**Comment by sincere-music** [comment]
I have been getting this issue since yesterday on Lubuntu 22.04 with Qt 5.15.3. Is there a known workaround short of avoiding LXQt?
I’ll see if behaviour changes after disabling backlight change, as someone seems to have reported.

Seems like a fairly critical issue, with this being an LTS distro.
